### ISSUE TITLE
Date type

### ISSUE THREAD

**Issue by shabzo** [sqlitebrowser/sqlitebrowser#2392]
Details for the issue
--------------------

#### What did you do?
Updated from visual studio date 2020-09-07

#### What did you expect to see?
2020-09-07

#### What did you see instead?
2020-09-07 00:00:00

Useful extra information
-------------------------

The info below often helps, please fill it out if you're able to. :)

#### What operating system are you using?

- [ ] Windows: ( _version:_ __7_ )
- [ ] Linux: ( _distro:_ ___ )
- [ ] macOS: ( _version:_ ___ )
- [ ] Other: ___

#### What is your DB4S version?
3.12.0
- [ ] 3.12.0
- [ ] 3.11.x
- [ ] Other: ___

#### Did you also

- [ ] Try out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [ ] Search for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by justinclift** [comment]
Any chance you could include a screenshot?

Just asking because it's a bit hard to visualise how it's going wrong, at least for me.  That being said, someone else might see this and know exactly what you mean and how to fix it. :wink:

**Comment by chrisjlocke** [comment]
'Updated from Visual Studio' ?  Can you give more details?
How are you storing '2020-09-07' ?  Its either a string (or text) value, or a numerical value.
Regardless of how its stored, you have a way of viewing that data how you want.  

SQLite doesn't technically have a 'datetime' datatype - its either text or a number.

**Comment by shabzo** [comment]
Thank you all for your response. Sorry for the late reply.
I had the date time picker format set to custom in form properties. Then
set custom format to yyyy-MM-dd.

**Comment by chrisjlocke** [comment]
Glad you got it resolved.

**Comment by shabzo** [comment]
Hi Chris

I have a problem. Im using a datetime picker in my visual studio form c#.
Ite format to yyyy-Mm-dd. I seemt to inserr delete fine from my forms. I am
having a problem with select * from t1 to fill datagridvuew1. I am getting
an error message datetime not correct. I know sqlite uses date. Please help

**Comment by shabzo** [comment]
Basicalky wat im trying to do. If i need to edit a sale i am calling the
tabke to make amends. Obviuisly the date is the problem.

**Comment by justinclift** [comment]
@shabzo Chris is in the UK, so it'll probably be a few hours until he's around... :wink:

**Comment by shabzo** [comment]
Ok. Thanks for letting me know

**Comment by chrisjlocke** [comment]
What SQL are you using?
Just use `select * from t1 where fieldName='2020-09-08'` where the date is in yyyy-MM-dd format.  Or you can do `select * from t1 where fieldName>'2020-09-08'` etc.  The date field is just a text field.  
I use VB, not c#, but the guts are the same.  So you'd put in code: `select * from t1 where fieldName='{dtPicker.value:yyyy-MM-dd}'`

If you need to edit a sale, then that record would have a unique ID, so you need to edit based on that... but I assume you're displaying a set of records which the user can then edit...

**Comment by shabzo** [comment]
Yes it worked... thank you so much..

Need some advice... does the client need sqlite db browser installed on
their pc. If they arw updating or editing from a form in visual studio win
app. Do i need to include the installation in the setup.

Thanks

**Comment by chrisjlocke** [comment]
No, DB4S (or DB Browser for SQLite) is just an application to build/edit/maintain the SQLite databases.
Theoretically, all you need is the files in the 'debug' directory.  Are you using the system.data.sqlite wrapper? (link above).  If so, you just need SQLite.Interop.dll (in two directories) and System.Data.SQLite.dll in the main directory, but this should all be in the debug directory anyway.  So you'd have
bin\debug\application.exe
bin\debug\database.db
bin\debug\System.Data.SQLite.dll
bin\debug\x86\SQLite.Interop.dll
bin\debug\x64\SQLite.Interop.dll

Obviously, the database can live where you want, but the DLL files have to be in the app directory and the x86\x64 directories.

**Comment by shabzo** [comment]
You sych a great help. Thank you

**Comment by chrisjlocke** [comment]
That of course is mirrored (or should do) if you use bin\release\* - I don't tend to, and just release the debug version...

**Comment by chrisjlocke** [comment]
The downloads for system.data.sqlite are here: https://system.data.sqlite.org/index.html/doc/trunk/www/downloads.wiki
However, there are so many downloads, its not overly clear what you need.
Download the 64-bit version (https://system.data.sqlite.org/downloads/1.0.113.0/sqlite-netFx46-binary-x64-2015-1.0.113.0.zip) and just use SQLite.Interop.dll - that goes in the x64 directory.
Download the 32-bit version (https://system.data.sqlite.org/downloads/1.0.113.0/sqlite-netFx46-binary-Win32-2015-1.0.113.0.zip) and just use SQLite.Interop.dll - that goes in the x86 directory.  Also use System.Data.SQLite.dll from that archive which goes in the bin\debug directory.  In your project, you just reference the System.Data.SQLite.dll.

**Comment by shabzo** [comment]
Hi

Will it be okay to send you my visual studio 2008 windows applicatio and
the database file. I dont k ow how to explain what i wnt. So it would be
easier if you then can try it out.

Please confirm.


Thanking you

**Comment by chrisjlocke** [comment]
> I dont k ow how to explain what i wnt.

Excellent! ;)  I assume you have a general idea.  You said about editing a sale.  Whats the general workflow?
Each sale is linked to a customer?  

Is this a database you're written, or you're creating an application around an existing database, or are you building everything yourself?

You said you're using Visual Studio, but in c# or vb?

**Comment by shabzo** [comment]
Hi

Yes. Each sale us linked to a custoner..using c# windows application.

Its a rental tyoe of setup.
PROBLEMS I AM HAVING..

i have to run a monthend with a zero balance for each customer and for the
12 items. Reason for this is to get the last day of the month as a sale so
that the days can be worked out..

If for example a client with a 500 cystomer base is gonna have another 500
rows of zero bal on the database and on the report.

Im sending you the solution file and the database.   Password for login is
user vinishree password 9304... and for user 1 it is user1@123

Please check it out and maybe you can find an alternative.. anotuer thing
on the report if an amount is 10 the report adda a 1 in front to display
110 and if 30 it displays 330... tue database is hust a mock for now...

Your help will be highly appreciated

**Comment by shabzo** [comment]
Basically if a customer buys 1 cyl and returns an empty there us no rental
charge. If he buys 5 and returns 2.then the rental charge until his next
purchase is 3 cyl per day. The next date on the report is the next
transaction for the custo.er otherwise it defaults to last day of that mo
th.

**Comment by shabzo** [comment]
fortest.rar
<https://drive.google.com/file/d/15FbuZd0b1_VmdUcomkUuuIcAjuyGLWBI/view?usp=drive_web>
